libs/libs-pomfix/lib/fix.js

```diff
@@ -0,0 +1,197 @@
+var es = require('event-stream')
+var combine = require('stream-combiner')
+
+var notNull = function (o) { return o !== null }
+var notUndefined = function (o) { return o !== undefined }
+
+var processPom = function ($) {
+  var descriptor = $('plugin descriptor')
+    .map(function (i, el) {
+      var descriptor = $(el).text().trim()
+      var pattern = /mvn:(.*)\/(.*)\/(.*)\/xml\/features/
+      var match = descriptor.match(pattern)
+
+      if (match !== null) {
+        return {
+          group: match[1],
+          artifact: match[2],
+          version: match[3],
+          type: 'xml',
+          classifier: 'features'
+        }
+      }
+    }).get()
+
+  var artifactItem = $('plugin artifactItem')
+    .map(function (i, el) {
+      var value = function (tag) {
+        return $(el).find(tag).first().text().trim() || undefined
+      }
+
+      return {
+        group: value('groupId'),
+        artifact: value('artifactId'),
+        version: value('version'),
+        scope: value('scope'),
+        type: value('type')
+      }
+    }).get()
+
+  return [].concat(descriptor).concat(artifactItem)
+}
+
+var processFeature = function ($) {
+  var pattern = /.*mvn:([^/]+)\/([^/]+)\/([^/]+)(\/)?(.+)?/
+
+  // find the implicit dependencies in the features.xml file
+  return $('feature bundle')
+    .map(function (i, el) {
+      var bundle = $(el).text().trim().match(pattern)
+
+      return {
+        group: bundle[1],
+        artifact: bundle[2],
+        version: bundle[3],
+        type: bundle[5]
+      }
+    }).get()
+}
+
+var makeDep = function ($) {
+  var makeTag = function (tag) { return $('<' + tag + '>') }
+  return function (dep) {
+    var required = [
+      makeTag('groupId').text(dep.group),
+      makeTag('artifactId').text(dep.artifact)
+    ]
+
+    var optional = ['version', 'scope', 'classifier', 'type', 'optional']
+      .map(function (tag) {
+        if (dep[tag] !== undefined) {
+          return makeTag(tag).text(dep[tag])
+        }
+      }).filter(notUndefined)
+
+    return makeTag('dependency').append(required.concat(optional))
+  }
+}
+
+module.exports = function (load, options) {
+  var opts = options || {}
+
+  // filter out unneeded dependencies
+  var keep = function (dep) {
```

Comment:
Because naming things is hard lol. How about `exclude(dep)` so I can do `.filter(exclude)`?